Code from demo does not work

I am following the video of the demo (which is excellent, thanks a lot!). I am at the step "Improve the layout", which says:
*To make the list of entries look like a blog, replace `<div class="table-responsive">` with HTML, so it uses a stacked layout in a single column.*

The code that is listed in the README does not work for me though: the replacement code uses `entries`, which are called `posts` in my original code. Then, there is a problem with `entry.date`, where IntelliJ reports "*Argument type dayjs.Dayjs is not assignable to parameter type Date | string | number*". Also, `entry.log.user` cannot be resolved, which is strange given this relationship is defined in the jdl file. Any help appreciated.

Unfortunately, I am getting this error now:
```
[INFO] Error: src/main/webapp/app/entities/post/list/post.component.html:37:82 - error TS2339: Property 'user' does not exist on type 'Pick<IBlog, "id" | "name">'.
[INFO] 
[INFO] 37         <small>Posted on {{ post.date | formatMediumDatetime }} by {{ post.blog?.user?.login }}</small>
```
despite class `Blog` having the corresponding properties/methods:
```
    @ManyToOne
    private User user;

   public User getUser() {
        return this.user;
    }
```

--- a/README.adoc
+++ b/README.adoc
@@ -414,32 +414,26 @@ To make the list of entries look like a blog, replace `<div class="table-respons
 [source,html]
 .src/main/webapp/app/entities/post/list/post.component.html
 ----
-<div class="table-responsive" *ngIf="entries?.length > 0">
-    <div infinite-scroll (scrolled)="loadPage(page + 1)" [infiniteScrollDisabled]="page >= links['last']" [infiniteScrollDistance]="0">
-        <div *ngFor="let entry of entries; trackBy: trackId">
-            <a [routerLink]="['/entry', entry.id, 'view' ]">
-                <h2>{{entry.title}}</h2>
-            </a>
-            <small>Posted on {{entry.date | date: 'short'}} by {{entry.blog.user.firstName}}</small>
-            <div [innerHTML]="entry.content"></div>
-            <div class="btn-group mb-2 mt-1">
-                <button type="submit"
-                        [routerLink]="['/entry', entry.id, 'edit']"
-                        class="btn btn-primary btn-sm">
-                    <fa-icon [icon]="'pencil-alt'"></fa-icon>
-                    <span class="d-none d-md-inline" jhiTranslate="entity.action.edit">Edit</span>
-                </button>
-                <button type="submit"
-                        [routerLink]="['/', 'entry', { outlets: { popup: entry.id + '/delete'} }]"
-                        replaceUrl="true"
-                        queryParamsHandling="merge"
-                        class="btn btn-danger btn-sm">
-                    <fa-icon [icon]="'times'"></fa-icon>
-                    <span class="d-none d-md-inline" jhiTranslate="entity.action.delete">Delete</span>
-                </button>
-            </div>
-        </div>
+<div class="table-responsive" *ngIf="posts && posts.length > 0">
+  <div infinite-scroll (scrolled)="loadPage(page + 1)" [infiniteScrollDisabled]="page >= links['last']" [infiniteScrollDistance]="0">
+    <div *ngFor="let post of posts; trackBy: trackId" data-cy="entityTable">
+      <a [routerLink]="['/post', post.id, 'view']" data-cy="entityDetailsButton">
+        <h2>{{ post.title }}</h2>
+      </a>
+      <small>Posted on {{ post.date | formatMediumDatetime }} by {{ post.blog?.user?.login }}</small>
+      <div [innerHTML]="post.content"></div>
+      <div class="btn-group mb-2 mt-1">
+        <button type="submit" [routerLink]="['/post', post.id, 'edit']" class="btn btn-primary btn-sm">
+          <fa-icon icon="pencil-alt"></fa-icon>
+          <span class="d-none d-md-inline" jhiTranslate="entity.action.edit" data-cy="entityEditButton">Edit</span>
+        </button>
+        <button type="submit" (click)="delete(post)" class="btn btn-danger btn-sm" data-cy="entityDeleteButton">
+          <fa-icon icon="times"></fa-icon>
+          <span class="d-none d-md-inline" jhiTranslate="entity.action.delete">Delete</span>
+        </button>
+      </div>
     </div>
+  </div>
 </div>
 ----
 

